### ISSUE TITLE
lammps Python binding: documentation needs update

### ISSUE THREAD

**Issue by elcorto** [mala-project/mala#455]
The instructions on how to build the lammps Python module are a bit brief for people not used to building lammps from source.

## Compiler 

One needs to

* (probably) use the same C compiler that was used to built Python
* have an MPI-aware compiler since [the docs say `make mode=shlib mpi`](https://mala-project.github.io/mala/install/external_modules.html#lammps-descriptor-calculation)

So something like

```sh
$ ml gcc/7.3.0
$ ml openmpi/4.1.1
```

if 

```sh
$ python -VV
Python 3.11.2 (main, Mar 10 2023, 09:40:18) [GCC 7.3.0]
```

## Using lammps' `install.py` script

Using the [`mala` branch](https://github.com/mala-project/lammps/tree/mala) (https://github.com/mala-project/lammps/commit/e16932716293dab2e0dbed8b666e7f9854b93a5f) of https://github.com/mala-project/lammps

```sh
$ python3 install.py -p lammps -l ../src/liblammps.so
```

fails

```sh
(mala-sys) schmer52@ga001:~/soft/git/mala-project/lammps/python$ python3 install.py -p lammps -l ../src/liblammps.so
usage: install.py [-h] -p PACKAGE -l LIB [-n] [-w WHEELDIR] -v VERSIONFILE
install.py: error: the following arguments are required: -v/--versionfile
```

One needs to add `-v ../src/version.h`.

**Comment by elcorto** [comment]
Also one needs to

* (probably) use the same C compiler that was used to built Python
* have an MPI-aware compiler since [the docs say `make mode=shlib mpi`](https://mala-project.github.io/mala/install/external_modules.html#lammps-descriptor-calculation)

So something like

```sh
$ ml gcc/7.3.0
$ ml openmpi/4.1.1
```

if 

```sh
$ python -VV
Python 3.11.2 (main, Mar 10 2023, 09:40:18) [GCC 7.3.0]
```

**Comment by elcorto** [comment]
As detailed in #456, the call to `install.py` should read

```sh
$ cd /path/to/lammps/python
$ python install.py -p lammps -l ../src/liblammps.so -v ../src/version.h -w $(pwd)
```

Replace `../src/liblammps.so` with `../build/liblammps.so` when using `cmake` instead of `make`.

**Comment by elcorto** [comment]
Re. compilers, here is a working example using `cmake` to build `liblammps.so` (HZDR hemera cluster module names). This also uses `-j32` jobs. We build an OpenMP version w/o MPI. 

```sh
$ ml python/3.11.2
$ python -VV
Python 3.11.2 (main, Mar 10 2023, 09:40:18) [GCC 7.3.0]

$ ml gcc/7.3.0 cmake/3.26.1

$ cd /path/to/lammps
$ mkdir build
$ cd build
$ cmake ../cmake/ -D BUILD_MPI=no -D BUILD_OMP=yes -D BUILD_SHARED_LIBS=yes -D PKG_ML-SNAP=yes
$ cmake --build . -j32 --clean-first

$ cd ../python/
$ python install.py -p lammps -l ../build/liblammps.so -v ../src/version.h -w $(pwd)
```

**Comment by elcorto** [comment]
`-w $(pwd)` is fixed on lammps side and is not needed any longer.

**Comment by RandomDefaultUser** [comment]
Will be closed by #467
